src/App.tsx

```diff
@@ -1,14 +1,43 @@
+/* eslint-disable @typescript-eslint/no-shadow */
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import { getTodos } from './api';
+
+import { Todo } from './types/Todo';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoId, setTodoId] = useState(0);
+  const [status, setStatus] = useState('');
+  const [input, setInput] = useState('');
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos);
```

Comment:
Let’s use async await. It gives you a simpler way to work with asynchronous promise-based code